docs/adr/0003-outsource-user-management.md

```diff
@@ -0,0 +1,89 @@
+# 3. Outsource User Management
+
+* Status: accepted <!-- optional -->
+* Deciders: @butonic, @micbar, @dragotin, @hodyroff, @pmaier1 <!-- optional -->
+* Date: 2020-12-09 <!-- optional -->
+
+Technical Story: [Skip account-service by talking to CS3 user-api](https://github.com/owncloud/ocis/pull/1020) <!-- optional -->
+
+## Context and Problem Statement
+
+To attach metadata like shares to users ownCloud relies on persistent, non-reassignable, unique identifiers for users (and files). Email and username can change when a user changes his name. But even the OIDC sub+iss combination may change when the IdP changes. While there is [an account porting protocol](https://openid.net/specs/openid-connect-account-porting-1_0.html) that describes how a relying party such as ownCloud should should behave, it still requires the RP to maintain its own user identifiers.
+
+## Decision Drivers <!-- optional -->
+
+* OCIS should be a single binary that can run out of the box without external dependencies like an LDAP server.
+* Time: we want to build a release candidate asap.
+
+## Considered Options
+
+* Accounts service wraps LDAP
+* GLauth wraps accounts service
+
+## Decision Outcome
+
+Chosen option: "Move accounts functionality to GLauth and name it accounts", by moving the existing accounts service file based persistence to glauth and use it as a drop in replacement for an LDAP server. The reverse index and web ui existing in the accounts service will move as well in order to make glauth a standalone, small scale user management with write capabilities.
+
+### Product summary
+- GLauth is a drop in user management for small scale deployments.
+- OCIS admins can either use the web ui to manage users in glauth or use existing tools in their IDM.
+- We hide the complexity by embedding OpenID Provider, an LDAP server and a user management web ui.
+
+### Resulting deployment options
+- Single binary: admin can manage users, groups and roles using the built in web ui (glauth)
+- External LDAP: OCIS admin needs to use existing tool to manage users
+- Separate OCIS and LDAP admin: OCIS admin relies on the LDAP admin to manage users
+
+### Resulting technical implications
+- add graphapi to glauth so the ocis web ui can use it to manage users
+- make graphapi service to directly talk to an LDAP server so our web ui can use it
+- keep the accounts service but embed glauth
+- add graph api to the accounts service?
```

Comment:
Is this section (`Resulting technical implications`) somehow a leftover? It sounds like `technical options - pick one` not like implications of the decision.

Reply:
It could be improved, yes ... and you had good questions. I'll try to incorporate my explanation.

Reply:
reworked